File: HydraScan/HydraScan/Models/Outcome.swift
import Foundation

enum OutcomeActor: String, Codable, CaseIterable, Hashable, Identifiable {
    case client
    case practitioner

    var id: String { rawValue }
}

enum TriStateChoice: String, Codable, CaseIterable, Hashable, Identifiable {
    case yes
    case maybe
    case no

    var id: String { rawValue }
}

enum RepeatIntent: String, Codable, CaseIterable, Hashable, Identifiable {
    case yes
    case maybe
    case no
    case noTryDifferent = "no_try_different"

    var id: String { rawValue }
}

struct Outcome: Identifiable, Codable, Hashable {
    var id: UUID
    var sessionID: UUID
    var clientID: UUID
    var clinicID: UUID?
    var recordedBy: OutcomeActor
    var recordedByUserID: UUID?
    var stiffnessBefore: Int?
    var stiffnessAfter: Int?
    var sorenessBefore: Int?
    var sorenessAfter: Int?
    var mobilityImproved: TriStateChoice?
    var sessionEffective: TriStateChoice?
    var readinessImproved: TriStateChoice?
    var repeatIntent: RepeatIntent?
    var romAfter: [String: Double]
    var romDelta: [String: Double]
    var clientNotes: String?
    var practitionerNotes: String?
    var createdAt: Date
}

File: shared/src/types/outcome.ts
export type OutcomeActor = "client" | "practitioner";

export type TriStateChoice = "yes" | "maybe" | "no";

export type RepeatIntent = "yes" | "maybe" | "no" | "no_try_different";

export interface OutcomeRecord {
  id: string;
  session_id: string;
  client_id: string;
  clinic_id: string;
  recorded_by: OutcomeActor;
  recorded_by_user_id: string;
  stiffness_before: number | null;
  stiffness_after: number | null;
  soreness_before: number | null;
  soreness_after: number | null;
  mobility_improved: TriStateChoice | null;
  session_effective: TriStateChoice | null;
  readiness_improved: TriStateChoice | null;
  repeat_intent: RepeatIntent | null;
  rom_after: Record<string, number> | null;
  rom_delta: Record<string, number> | null;
  client_notes: string | null;
  practitioner_notes: string | null;
  created_at: string;
  updated_at?: string;
}
